File: app/src/main/java/com/example/mobileappdevelopment/DataUtils/Cache.java
package com.example.mobileappdevelopment.DataUtils;

import com.example.mobileappdevelopment.Model.Hunt;
import com.google.android.gms.auth.api.signin.GoogleSignInAccount;
import com.google.android.gms.tasks.Task;
import com.google.firebase.firestore.QuerySnapshot;

import java.util.ArrayList;
import java.util.List;

public class Cache {

    public static List<Hunt> allHunts = new ArrayList<>();
    public static List<String> allHuntTitles = new ArrayList<>();
    public static List<String> allHuntAuthors = new ArrayList<>();

    public static boolean isPrivate = false;
    public static boolean listUpdated = false;

    public static GoogleSignInAccount account = null;

    public static Task<QuerySnapshot> query = null;
}

File: app/src/main/java/com/example/mobileappdevelopment/DataUtils/Cache.kt
package com.example.mobileappdevelopment.DataUtils

import com.example.mobileappdevelopment.Model.Hunt
import com.google.android.gms.auth.api.signin.GoogleSignInAccount
import com.google.android.gms.tasks.Task
import com.google.firebase.firestore.QuerySnapshot
import java.util.*
import kotlin.collections.ArrayList

object Cache {
    var allHunts: MutableList<Hunt> = ArrayList()
    var allHuntTitles: MutableList<String> = ArrayList()
    var allHuntAuthors: MutableList<String> = ArrayList()
    var isPrivate = false
    var listUpdated = false
    var account: GoogleSignInAccount? = null
    var query: Task<QuerySnapshot>? = null
}